presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/IterativeOptimizer.java

```diff
@@ -58,6 +61,8 @@
     private final List<PlanOptimizer> legacyRules;
     private final RuleIndex ruleIndex;
     private final Optional<LogicalPropertiesProvider> logicalPropertiesProvider;
+    private final String optimizerName;
+    private final Set<String> rulesTriggered;
```

Comment:
instead of collecting rules triggered and putting them under a single optimizer name, i think it would be better to consider each rule as an optimizer, and add them as we go through.

Reply:
Updated, all rules will have a separate entry in the logging.